### ISSUE TITLE
Validate K-branch theta arity; malformed theta causes probability loss and backend divergence

### ISSUE THREAD

**Issue by haasonsaas** [extropic-ai/torx#11]
## Severity

**High** — wrong theta length corrupts probability mass and makes StateVector and Branching simulators disagree on the same model.

## Affected version

- PyPI `extro-torx==0.0.1`
- Commit `769d2f90abdfda14798fceb521143f4b99d370da`

## Summary

A K-branch gate requires exactly `K - 1` logits, but this is never enforced.

For `K > 2`, `probs()` performs softmax over every supplied theta value. Concrete failure:

```python
gate = PditCycle(0, dims=3)  # 3 branches, requires 2 logits
matrix = gate.get_matrix(jnp.zeros(3))  # supplied 3 logits
```

That creates four probabilities of `0.25`, but `PditCycle.get_matrix()` consumes only the first three. Every transition-matrix column sums to `0.75`.

Missing parameters are also dangerous: with one theta instead of two, column sums can become `1.5`.

Meanwhile the branch compiler silently slices extra logits:

```python
# torx/psc/simulation/sampled.py
padded_theta.at[: K - 1].set(theta_vals[: K - 1])
```

## Verified on pinned commit

| Backend | theta length 3 (should be 2) | Result |
|---------|------------------------------|--------|
| StateVector | density sum | **0.75** |
| Branching | empirical sum (sliced to 2 logits) | **1.0** |

The same parameter list means different models in different simulators.

## Suggested fix

Reject any static theta shape other than `(gate.num_branches - 1,)` in every compiler. Assert every generated matrix is finite, nonnegative, and column-stochastic.

## Regression tests

- Cross-backend: exact probabilities vs large-sample branching estimates on small valid circuits.
- Malformed theta lengths rejected at compile/build time.
- Stochastic-matrix property checks over a theta grid for built-in discrete gates.

**Comment by lockwo** [comment]
yes we do not check every input. if this becomes a common pain point among users, we can potentially add a check, but we don't want to do so prematurely
